### PR TITLE
Replace activity `ACT_ADV_INVENTORY` with `uistate.open_menu`, remove `ACT_VIEW_RECIPE`

### LINKED ISSUE #62996: `view recipe` works only once per opened Advanced Inventory
### Describe the bug

Can `view recipe` of "in progress item" in advanced inventory only once, until Advanced inventory is closed and opened again.

### Steps to reproduce

Prep:
1. load my save (see additional context)

Bug:
1. Open Advanced inventory (`/`)
2. scroll to "in progress item", like `progress hotplate (55%)`
3. view actions (`e`)
4. in opened menu `viewing recipe` (`V`).
5. Crafting menu opens with expected item.
6. `ESC` to close crafting **don't close advanced inventory**
7. Until you close inventory, no "in progress item" opens anymore, instead the actions menu (step 3) closes.

Summary: load save; press `/eV` `ESC` `eV` 

If you close and open advanced inventory, you can view one "in progress item" again given steps 1. to 4.

### Expected behavior

Can inspect as many "in progress items" as I want.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2364 (21H2)
- Game Version: a00a2ec [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

I expect my reproduction to work reliably, but I am lazy, so I am attaching my save.
save: [Ransom Canyon.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/10348439/Ransom.Canyon.zip)

Also, I will try to fix it now :)

### PR DESCRIPTION
**Pull request by Brambor**
#### Summary
None

#### Purpose of change

- Fix #62996
- Using "View recipe" in AIM no longer takes an in-game second. It takes no in-game time, as opening a menu should.
#### Describe the solution

1. Stare at the code for _two whole days_.
2. Remove `ACT_VIEW_RECIPE`, just run the menu code directly, as it is just opening a menu over AIM, which is fine.
3. Replace activity `ACT_ADV_INVENTORY` with `uistate.open_menu = create_advanced_inv`.
4. Remove legacy code and migrate activity to null.

#### Describe alternatives you've considered

They were all stupid.

I could also make this as two PRs. There would be a conflict on the save_legacy though.

#### Side effect

AIM doesn't open when it should after the save/load cycle if autosave is triggered during the "return to AIM" activity - see testing. This is fine.

#### Testing

**Test fix**:
 - See the reproduction steps in #62996

**Test temp_hide_advanced_inv**
1. have a lighter
2. in AIM examine > activate
3. The AIM hid, as expected

**Test migration**
What I did:

1. Load the save in the old version.
4. Set saving interval to 0 min and save after 10 actions.
5. `/` Open AIM.
6. `e`xamine machete.
7. `V`iew recipe. (repeat 10 times)
   - The autosave will save during the activity.
8. alt+F4

There is the save:
[TEST migration AIM & VIEW.zip](https://github.com/user-attachments/files/17020013/TEST.migration.AIM.VIEW.zip)

Note it has `ACT_ADV_INVENTORY` and `ACT_VIEW_RECIPE` in it in the .sav file

On master:
1. Load save.
2. It opens AIM after loading.

In the new version:

1. Load save.
2. AIM doesn't open and the game doesn't complain about missing `ACT_ADV_INVENTORY` or `ACT_VIEW_RECIPE`.

### CODE REVIEW

**Comment by mqrause** [comment]
Are you interested in giving `ACT_PICKUP_MENU` the same treatment (if it's possible)?

**Comment by Brambor** [comment]
> Are you interested in giving `ACT_PICKUP_MENU` the same treatment (if it's possible)?

Yup, I will go through them and see what I can do!

It is more time-consuming than it should be, as I try to ensure I didn't break anything...

**Comment by Brambor** [comment]
I realized I didn't test temp_hide, so I will do that tomorrow.

File: src/character.cpp [hunk 1]
```diff
@@ -8842,19 +8840,6 @@ void Character::cancel_activity()
...
-    bool has_adv_inv = has_activity( ACT_ADV_INVENTORY );
-    for( auto backlog_item = backlog.begin(); backlog_item != backlog.end(); ) {
-        if( backlog_item->auto_resume &&
-            ( !has_adv_inv || backlog_item->id() != ACT_ADV_INVENTORY ) ) {
-            has_adv_inv |= backlog_item->id() == ACT_ADV_INVENTORY;
-            backlog_item++;
-        } else {
-            backlog_item = backlog.erase( backlog_item );
-        }
-    }
```
[the remark this answers is not in the document]
Comment (Brambor) [reply src/character.cpp:8857]: Good question, let me explain...

... I rechecked myself three times before making that change and still made a mistake! Good catch!

[the remark this answers is not in the document]
Comment (Brambor) [reply src/character.cpp:8857]: Thank you, I will fix it. Also, I will look around to see what else I can ax.

**Comment by PatrikLundell** [comment]
I'm all in favor of spending time to ensure things aren't broken...

**Comment by Brambor** [comment]
@andrei8l Do you know if this will still be needed?

https://github.com/CleverRaven/Cataclysm-DDA/blob/d5e57fa771d295561933f6c64951050b22cb4792/src/do_turn.cpp#L534-L537

My quick testing at removing it didn't break anything obvious...

**Comment by Brambor** [comment]
I don't plan to work on it then. However, there is the patch:

```patch
commit 68af3324a0e5c074039f661a4ab3a0fd0e037ac7
Author: Brambor <tondadrd@seznam.cz>
Date:   Tue Sep 17 14:35:25 2024 +0200

    TODO remove unneded ? kill_advanced_inv

diff --git a/src/advanced_inv.cpp b/src/advanced_inv.cpp
index 44fa35ae8a..235c4d2418 100644
--- a/src/advanced_inv.cpp
+++ b/src/advanced_inv.cpp
@@ -117,6 +117,12 @@ namespace
 std::unique_ptr<advanced_inventory> advinv;
 } // namespace
 
+void kill_advanced_inv()
+{
+    advinv.reset();
+    cancel_aim_processing();
+}
+
 void create_advanced_inv()
 {
     if( !advinv ) {
@@ -129,12 +135,6 @@ void create_advanced_inv()
     }
 }
 
-void kill_advanced_inv()
-{
-    advinv.reset();
-    cancel_aim_processing();
-}
-
 void temp_hide_advanced_inv()
 {
     if( advinv ) {
diff --git a/src/do_turn.cpp b/src/do_turn.cpp
index 21ff4c985a..64fde0e095 100644
--- a/src/do_turn.cpp
+++ b/src/do_turn.cpp
@@ -531,10 +531,6 @@ bool do_turn()
     while( u.get_moves() > 0 && u.activity ) {
         u.activity.do_turn( u );
     }
-    // FIXME: hack needed due to the legacy code in advanced_inventory::move_all_items()
-    if( !u.activity ) {
-        kill_advanced_inv();
-    }
 
     // Process NPC sound events before they move or they hear themselves talking
     for( npc &guy : g->all_npcs() ) {
diff --git a/src/ui.h b/src/ui.h
index ce63b0b826..780935b6c6 100644
--- a/src/ui.h
+++ b/src/ui.h
@@ -553,7 +553,6 @@ class pointmenu_cb : public uilist_callback
         void select( uilist *menu ) override;
 };
 
-void kill_advanced_inv();
 void temp_hide_advanced_inv();
 
 /**

```

**Pull request merged**